packages/blade/src/components/ActionList/ActionListItem.tsx

```diff
@@ -0,0 +1,235 @@
+
+const ActionListItemIcon = ({ icon }: { icon: IconComponent }): JSX.Element => {
+  const Icon = icon;
+  const { intent } = React.useContext(ActionListItemContext);
+  return (
+    <Icon
+      color={
+        intent === 'negative'
+          ? 'feedback.icon.negative.lowContrast'
+          : 'surface.text.muted.lowContrast'
+      }
+      size="medium"
+    />
+  );
+};
+
+const ActionListItemText = ({ children }: { children: string }): JSX.Element => {
+  return (
+    <BaseText fontStyle="italic" fontSize={50} color="surface.text.muted.lowContrast">
+      {children}
+    </BaseText>
+  );
+};
+
+const getActionListItemRole = (href?: string): 'link' | 'menuitem' | 'option' => {
+  if (href) {
+    return 'link';
+  }
+
+  if (isReactNative()) {
+    return 'menuitem';
+  }
+
+  return 'option';
+};
+
+/**
+ *
+ * ActionListItem
+ *
+ *
+ */
+type ActionListItemProps = {
+  title: string;
+  description?: string;
+  onClick?: (e: React.MouseEvent<HTMLButtonElement>) => void;
+  value: string;
+  href?: string;
+  /**
+   * Internally passed from ActionList. No need to pass it explicitly
+   */
+  index?: number;
+  leading?: React.ReactNode;
+  trailing?: React.ReactNode;
+  isDefaultSelected?: boolean;
+  intent?: 'negative';
+};
+
+type ClickHandlerType = (e: React.MouseEvent<HTMLButtonElement>) => void;
+
+const makeActionListItemClickable = (
+  clickHandler: ClickHandlerType,
+): { onPress?: StyledActionListItemProps['onPress']; onClick?: ClickHandlerType } => {
+  if (isReactNative()) {
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment, @typescript-eslint/prefer-ts-expect-error
+    // @ts-ignore: ignoring ReactNative press type for the peace of mind
+    return { onPress: clickHandler };
+  }
+
+  return {
+    onClick: clickHandler,
+  };
+};
+
+const ActionListItem = (props: ActionListItemProps): JSX.Element => {
+  const {
+    activeIndex,
+    dropdownBaseId,
+    onOptionClick,
+    selectedIndices,
+    setShouldIgnoreBlur,
+    selectionType,
+    selectInputRef,
+  } = useDropdown();
+
+  const renderOnWebAs = props.href ? 'a' : 'button';
+  const isSelected =
+    typeof props.index === 'number'
+      ? selectedIndices.includes(props.index)
+      : props.isDefaultSelected;
+
+  return (
+    <ActionListItemContext.Provider value={{ intent: props.intent }}>
+      <StyledActionListItem
+        as={!isReactNative() ? renderOnWebAs : undefined}
+        id={`${dropdownBaseId}-${props.index}`}
+        tabIndex={-1}
+        href={props.href}
+        className={activeIndex === props.index ? 'active-focus' : ''}
+        {...makeAccessible({
+          selected: isSelected,
+          role: getActionListItemRole(props.href),
+        })}
+        {...makeActionListItemClickable((e: React.MouseEvent<HTMLButtonElement>): void => {
+          if (typeof props.index === 'number') {
+            onOptionClick(e, props.index);
+          }
+          props.onClick?.(e);
+        })}
```

Comment:
Is the order matters here? 

If not, can we do this interpolated props at the end of all other props? (bit cleaner to look at)

Reply:
I have kept the custom properties on styled component at the end and native HTML attributes at the start. So that its easier to debug for custom properties.